cdc/sinkv2/eventsink/txn/mysql/mysql.go

```diff
@@ -0,0 +1,543 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+	"net"
+	"net/url"
+	"sort"
+	"strconv"
+	"time"
+
+	dmysql "github.com/go-sql-driver/mysql"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/parser/charset"
+	"github.com/pingcap/tidb/parser/mysql"
+	"go.uber.org/zap"
+
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/sinkv2/eventsink"
+	"github.com/pingcap/tiflow/cdc/sinkv2/metrics"
+	dmutils "github.com/pingcap/tiflow/dm/pkg/utils"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"github.com/pingcap/tiflow/pkg/quotes"
+	"github.com/pingcap/tiflow/pkg/retry"
+	"github.com/pingcap/tiflow/pkg/sink"
+)
+
+const (
+	backoffBaseDelay = 500 * time.Millisecond
+	backoffMaxDelay  = 60 * time.Second
+
+	// Max interval for flushing transactions to the downstream.
+	maxFlushInterval = 100 * time.Millisecond
+
+	defaultDMLMaxRetry uint64 = 8
+
+	// defaultDDLMaxRetry uint64 = 20
+)
+
+type mysqlBackend struct {
+	changefeedID model.ChangeFeedID
+	db           *sql.DB
+	params       *sinkParams
+	options      sinkOptions
+	statistics   *metrics.Statistics
+
+	events []*eventsink.TxnCallbackableEvent
+	rows   int
+
+	cancel func()
+}
+
+// NewMySQLBackend creates a new MySQL sink using schema storage
+func NewMySQLBackend(
+	ctx context.Context,
+	changefeedID model.ChangeFeedID,
+	sinkURI *url.URL,
+	opts ...sinkOptions,
+) (*mysqlBackend, error) {
+	if len(opts) == 0 {
```

Comment:
What happens in the case? If we always pass in config, we use whatever value it is.